lib/ams_lazy_relationships/core/evaluation.rb

```diff
@@ -13,51 +13,55 @@ module Evaluation
     #
     # @param object [Object] Lazy relationships will be loaded for this record.
     # @param level [Integer] Current nesting level
-    def load_all_lazy_relationships(object, level = NESTING_START_LEVEL)
+    def load_all_lazy_relationships(object, scope, level = NESTING_START_LEVEL)
       return if level >= LAZY_NESTING_LEVELS
       return unless object
 
       return unless lazy_relationships
 
       lazy_relationships.each_value do |lrm|
-        load_lazy_relationship(lrm, object, level)
+        load_lazy_relationship(lrm, object, scope, level)
       end
     end
 
     # @param lrm [LazyRelationshipMeta] relationship data
     # @param object [Object] Object to load the relationship for
     # @param level [Integer] Current nesting level
-    def load_lazy_relationship(lrm, object, level = NESTING_START_LEVEL)
+    def load_lazy_relationship(lrm, object, scope, level = NESTING_START_LEVEL)
       load_for_object = if lrm.load_for.present?
                           object.public_send(lrm.load_for)
                         else
                           object
                         end
 
-      lrm.loader.load(load_for_object) do |batch_records|
+      # Make sure that old custom loaders are still supported
+      loader_args = lrm.loader.method(:load).arity == 1 ? [load_for_object] : [load_for_object, scope]
+
+      lrm.loader.load(*loader_args) do |batch_records|
         deep_load_for_yielded_records(
           batch_records,
+          scope,
           lrm,
           level
         )
       end
     end
 
-    def deep_load_for_yielded_records(batch_records, lrm, level)
+    def deep_load_for_yielded_records(batch_records, scope, lrm, level)
       # There'll be no more nesting if there's no
       # reflection for this relationship. We can skip deeper lazy loading.
       return unless lrm.reflection
 
       Array.wrap(batch_records).each do |r|
-        deep_load_for_yielded_record(r, lrm, level)
+        deep_load_for_yielded_record(r, scope, lrm, level)
       end
     end
 
-    def deep_load_for_yielded_record(batch_record, lrm, level)
+    def deep_load_for_yielded_record(batch_record, scope, lrm, level)
       serializer = serializer_for(batch_record, lrm.reflection.options)
       return unless serializer
 
-      serializer.send(:load_all_lazy_relationships, batch_record, level + 1)
+      serializer.send(:load_all_lazy_relationships, batch_record, scope, level + 1)
```

Comment:
Probably we need to take scope from nested serializer, something like

```suggestion
      serializer.send(:load_all_lazy_relationships, batch_record, serializer.scope, level + 1)
```

Reply:
I don't think that's possible, because AFAIK `serializer` here is the class not the instance.
Btw the scope is in most cases (I haven't seen other :P) provided by the Rails controller by calling e.g. `serialization_scope :current_user`, so this should be fine.